nvdaHelper/vbufBackends/mshtml/mshtml.cpp

```diff
@@ -401,8 +402,11 @@ inline void getCurrentStyleInfoFromHTMLDOMNode(IHTMLDOMNode* pHTMLDOMNode, bool&
 	if (pHTMLCurrentStyle) pHTMLCurrentStyle->Release();
 }
 
+// #8976: the string in the following macro  must be passed to the COM method as a BSTR 
+// otherwise the COM marshaller will try and read the BSTR length and hit either inaccessible memory or get back junk. 
+// This is seen in optimized builds of NVDA when accessing some CHM files in hh.exe.
 #define macro_addHTMLAttributeToMap(attribName,allowEmpty,attribsObj,attribsMap,tempVar,tempAttrObj) {\

```

Comment:
I guess this has probably been a macro for a while. I'd be quite tempted to turn it into a regular function, the benefits of doing so is to make them more easy to debug, and to get rid of the backslashes.